File: problems/problems_3191/solution.ts
function minOperations(nums: number[]): number {
    const n: number = nums.length;
	let ans: number = 0;
	for (let i: number = 0; i < n - 2; i++) {
		if (nums[i] === 0) {
			ans++;
			nums[i + 1] ^= 1;
			nums[i + 2] ^= 1;
		}
	}
	return nums[n - 2] === 1 && nums[n - 1] === 1 ? ans : -1;
};

export function Solve(inputJsonElement: string): any {
	const inputValues: string[] = inputJsonElement.split("\n");
	const nums: number[] = JSON.parse(inputValues[0]);
	return minOperations(nums);
}

File: problems/problems_3191/solution.rs
use serde_json::{json, Value};

pub struct Solution;

impl Solution {
    pub fn min_operations(mut nums: Vec<i32>) -> i32 {
		let n = nums.len();
		let mut ans = 0;
		for i in 0..n-2 {
			if nums[i] == 0 {
				ans += 1;
				nums[i + 1] ^= 1;
				nums[i + 2] ^= 1;
			}
		}
		if nums[n - 2] == 0 || nums[n - 1] == 0 {
			return -1;
		}
		ans
    }
}

#[cfg(feature = "solution_3191")]
pub fn solve(input_string: String) -> Value {
	let input_values: Vec<String> = input_string.split('\n').map(|x| x.to_string()).collect();
	let nums: Vec<i32> = serde_json::from_str(&input_values[0]).expect("Failed to parse input");
	json!(Solution::min_operations(nums))
}
